server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/VisitorUtility.java

```diff
@@ -29,27 +28,25 @@
 @UtilityClass
 public class VisitorUtility {
   /**
-   * Construct locality
+   * Construct locality in extended document
    *
    * @param ctx
    * @param context
    * @return locality
    */
   public Locality constructLocality(ParserRuleContext ctx, DialectProcessingContext context) {
-    Location location = context.getExtendedDocument().mapLocation(constructRange(ctx));
-    return Locality.builder().uri(location.getUri()).range(location.getRange()).build();
+    return Locality.builder().uri(context.getExtendedDocument().getUri()).range(constructRange(ctx)).build();
```

Comment:
Issue is we try to adjust the dialect diagnostics . So these locality need not be adjusted.
But we don't seems to do the same on the dialect Nodes, which looks wrong. 
https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/f79f54352c8f271d2255d495d2ae4e57539fa554/server/engine/src/main/java/org/eclipse/lsp/cobol/dialects/ibm/ImplicitDialectProcessingStage.java#L84